docs/v1/P4Runtime-Spec.mdk

```diff
@@ -6269,6 +6279,314 @@ field [@ProtoAny]. At the moment, there is not any mechanism to extend the
+    * `DELETE` : Instead of deleting, this operation resets the entry with
+      default values for every field. The union field values are changed
+      but the union itself isn't changed.
+      Only Match fields must be set. `NOT_FOUND` is returned if
+      the entry isn't found. If no Match fields are given, then all entries
+      are reset (wildcard reset). If the table contained init entries, they
+      are restored to initial state with their original union and union field
+      values.
+      If `default_entry` is true, then the default entry is reset to initial
+      values. Const union rules apply.
+      Wildcard doesn't apply to default_entry.
+    * `READ` : The Read RPC. All entries can be read. If no entities are
+      provided in request, then all entries are read, AKA wildcard read.
+      If `default_entry` is true, then the default entry is returned. Match
+      fields should not be set in this case.
+      Wildcard read does NOT return default entry.
+  
+  * Read-only tables: Entries can only be read. Write RPC doesn't work.
+    * `READ` : The Read RPC. All entries can be read. If no entities are
+      provided in request, then all entries are read, AKA wildcard read.
+      If `default_entry` is true, then the default entry is returned. Match
+      fields should not be set in this case.
+      Wildcard read does NOT return default entry.
+
+  * Reset-only tables :
+    * `INSERT` : Not supported. Returns `NOT_SUPPORTED` error
+    * `MODIFY` : Not supported. Returns `NOT_SUPPORTED` error
+    * `DELETE` : Instead of deleting, this operation resets the entry with
+      default values for every field.
+      Only Match fields must be set. `NOT_FOUND` is returned if
+      the entry isn't found. If no Match fields are given, then all entries
+      are reset (wildcard reset). If the table contained init entries, they
+      are restored to initial state with their original union and union field
+      values.
+      If `default_entry` is true, then the default entry is reset to initial
+      values. Const union rules apply.
+      Wildcard doesn't apply to default_entry.
+    * `READ` : The Read RPC. All entries can be read. If no entities are
+      provided in request, then all entries are read, AKA wildcard read.
+      If `default_entry` is true, then the default entry is returned. Match
+      fields should not be set in this case.
+      Wildcard read does NOT return default entry.
+  
+  * Calculation tables: Entries can only be read. However different from
+    Read-only in the sense that there is no defined size of the table since
+    the possible range of matchfields can be immense. For example, hash
+    calculation tables where matchfields are the field values and the union
+    data contains the hash value. There is no default_entry for such tables.
+    * `READ` : The Read RPC. Wildcard read doesn't exist for this table.
+      Match fields should always be set
```

Comment:
Which error is returned if match fields are missing?
Which error is returned if an illegal operation is attempted?